core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/workflow/WorkflowResource.scala

```diff
@@ -477,12 +490,22 @@ class WorkflowResource {
       }
     }
 
-    // Apply owner filter
-    val ownerFilter = getOwnerFilter(owners)
     // combine all filters with AND
     var optionalFilters: Condition = noCondition()
     optionalFilters = optionalFilters
```

Comment:
The `optionalFilters` variable is never used. So, it doesn't seem to me that this would ever work....

It should have been caught by the unit tests. But the tests try to test internal methods, rather than exercising the entire /search method. i.e. the test should verify that given the correct parameters to the `WorkflowResource.searchWorkflows()` method, the correct result is returned. For this to work, the database has to be mocked. I believe we already set that up before.

In my opinion, `getDateFilter`, `getOwnerFilter`, etc should be private, although I see the point in testing them individually. Let's use the principle of encapsulation and create a `WorkflowSearchFilterGenerator` class where these helper methods are public. Then, `WorkflowResource` can call into it.